paasta_tools/paasta_metastatus.py

```diff
@@ -147,13 +176,84 @@ def utilization_table_by_grouping_from_mesos_state(
         table_rows.append(metastatus_lib.get_table_rows_for_resource_info_dict(
             [v for g, v in grouping_values],
             healthcheck_utilization_pairs,
-        ) + [str(resource_info_dict['slave_count'])])
+        ))
+
+        if service_instance_stats:
+            # Calculate the max number of runnable service instances given the current resources (e.g. cpus, mem, disk)
+            resource_free_dict = {rsrc.metric: rsrc.free for rsrc in resource_utilizations}
+            num_service_instances_allowed = float('inf')
+            limiting_factor = 'Unknown'
+            # service_instance_stats.keys() should be a subset of resource_free_dict
+            for rsrc_name, rsrc_amt_wanted in service_instance_stats.items():
+                if rsrc_amt_wanted > 0:
+                    # default=0 to indicate there is none of that resource
+                    rsrc_free = resource_free_dict.get(rsrc_name, 0)
+                    if rsrc_free // rsrc_amt_wanted < num_service_instances_allowed:
+                        limiting_factor = rsrc_name
+                        num_service_instances_allowed = rsrc_free // rsrc_amt_wanted
+            table_rows[-1].append('{:6} ; {}'.format(int(num_service_instances_allowed), limiting_factor))
+
+        # Always append the agent count last
+        table_rows[-1].append(str(resource_info_dict['slave_count']))
+
     table_rows = sorted(table_rows, key=lambda x: x[0:len(groupings)])
     all_rows.extend(table_rows)
 
     return all_rows, healthy_exit
 
 
+def get_service_instance_stats(service: str, instance: str, cluster: str) -> Dict[str, float]:
+    """Returns a Dict with stats about a given service instance.
+
+    Args:
+        service: the service name
+        instance: the instance name
+        cluster: the cluster name where the service instance will be searched for
+
+    Returns:
+        A Dict mapping resource name to the amount of that resource the particular service instance consumes.
+    """
+    if service is None or instance is None or cluster is None:
+        return {}
+
+    try:
+        instance_type: str = validate_service_instance(service, instance, cluster, DEFAULT_SOA_DIR)
+    except Exception as e:
+        log.error(f'Failed to get stats for service {service} instance {instance}:')
+        log.error(e)
+        return {}
+    # Keys match paasta_tools.utils.INSTANCE_TYPES
+    instance_class_dict = {
```

Comment:
Why can't we use `paasta_tools.cli.get_instance_config` here? It already has this lookup logic.

Reply:
Didn't know this was a thing but it made the code much more concise. Thanks!